### PR TITLE
Generate separate file for generated type models in Ruby by default

### PR DESCRIPTION
**Pull request by koesie10**
This will generate a separate file `models/ruby.model.generated.yml` when running the model editor for Ruby and type models are hidden (the default). This file is ignored by the model editor, but stores all type models generated by the query. This is part of hiding type models by default.

Screenshot 2024-02-23 at 15 55 38

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Comment by koesie10** [comment]
I'll change this to use the `codeQL.model.showTypeModels` setting instead of canary mode once https://github.com/github/vscode-codeql/pull/3399 is merged.

File: extensions/ql-vscode/src/model-editor/model-editor-view.ts [hunk 1]
```diff
@@ -731,7 +748,25 @@ export class ModelEditorView extends AbstractWebview<
...
+          return;
         }
+
+        progress({
+          step: 4000,
+          maxStep: 4000,
+          message: "Saving generated models",
+        });
+
+        const fileContents = `# This file was automatically generated based from ${this.databaseItem.name}. Manual changes will not persist.\n\n${modelExtensionFileToYaml(extensionFile)}`;
```
Comment (shati-patel) [inline extensions/ql-vscode/src/model-editor/model-editor-view.ts:760]: Minor suggestion:

```suggestion
        const fileContents = `# This file was automatically generated from ${this.databaseItem.name}. Manual changes will not persist.\n\n${modelExtensionFileToYaml(extensionFile)}`;
```

Or maybe change `from` to `on` or something like that instead?

File: extensions/ql-vscode/src/model-editor/generate.ts [hunk 2]
```diff
@@ -5,19 +5,15 @@ import type { QueryRunner } from "../query-server";
...
 import { resolveQueries } from "../local-queries";
 import type { DecodedBqrs } from "../common/bqrs-cli-types";
+
 type GenerateQueriesOptions = {
   queryConstraints: QueryConstraints;
   filterQueries?: (queryPath: string) => boolean;
-  parseResults: (
-    queryPath: string,
-    results: DecodedBqrs,
-  ) => ModeledMethod[] | Promise<ModeledMethod[]>;
```
Comment (shati-patel) [inline extensions/ql-vscode/src/model-editor/generate.ts:19]: Why are we getting rid of the `parseResults` option? 🤔

**Review by shati-patel — COMMENTED** [review]
Looks mostly good, just a few questions! 📝 

----------- 

I also had a general comment from testing: When I run `Generate` from the model editor, there's a log entry saying
```
CLI command succeeded.
No predicate found for typeVariableModel
``` 

I don't think it's necessarily related to your changes, but I also couldn't see exactly where it's coming from 😅 (This is on the nightly CLI version, and a recent `main` codeql submodule checkout—but I tried a few different versions too 🤔)

File: extensions/ql-vscode/src/model-editor/generate.ts (hunk 2 above)
Comment (koesie10) [reply extensions/ql-vscode/src/model-editor/generate.ts:19]: `parseResults` was very specific to producing `ModeledMethod[]`, while that's not something we want to do for this type model generation (we can just put the rows as-is into a YAML file). The only thing that `parseResults` was used for was `await onResults(await parseResults(queryPath, bqrs))`, which can very easily be replaced by `await onResults(queryPath, bqrs)` with very minor changes on the consumer side.

**Comment by koesie10** [comment]
> I don't think this "typeVariableModel" is necessarily related to your changes, but I also couldn't see exactly where it's coming from 😅 (This is on the nightly CLI version, and a recent `main` codeql submodule checkout—but I tried a few different versions too 🤔)

I believe this has been there since the beginning. The model editor doesn't support `typeVariableModel`, but it's a predicate that's produced from the generation query. We can ignore this log message.

**Review by shati-patel — APPROVED** [review]
Thanks for the explanations ✨ Looks good!

**Pull request merged**